packages/headless/src/features/commerce/context/cart/cart-actions-loader.ts

```diff
@@ -5,47 +5,85 @@ import {
   CommerceEngineState,
 } from '../../../../app/commerce-engine/commerce-engine';
 import {
+  CartActionPayload,
   PurchasePayload,
   SetItemsPayload,
   UpdateItemQuantityPayload,
+  emitCartAction,
+  emitPurchase,
   purchase,
   setItems,
   updateItemQuantity,
 } from './cart-actions';
 import {cartReducer as cart} from './cart-slice';
 
-export type {PurchasePayload, SetItemsPayload, UpdateItemQuantityPayload};
+export type {
+  CartActionPayload,
+  PurchasePayload,
+  SetItemsPayload,
+  UpdateItemQuantityPayload,
+};
 
 /**
  * The cart action creators.
  */
 export interface CartActionCreators {
   /**
-   * Emits an ec_purchase analytics events with the current cart state.
+   * Emits an `ec.purchase` analytics event with the current cart state.
+   *
+   * Should be dispatched before the `purchase` action.
    *
    * @param payload - The action creator payload.
+   * @returns A dispatchable action.
    */
-  purchase(
+  emitPurchase(
```

Comment:
Can we have access to this function (as an implementer)? I think it would be fine, but I would suggest to change the name (like `emitPurchaseEvent`), just to be clearer about the fact that this method will affect the analytic, not the state.

Reply:
It's worth noting that both names would be inconsistent with the product view controller where the method is called [view](https://github.com/coveo/ui-kit/blob/bff8876908fde6febaba8f9cadc0f23d199afc33/packages/headless/src/controllers/commerce/product-view/headless-product-view.ts#L13) instead of `emitView` or `emitViewEvent`. Similarly for the click, the method is called [select](https://github.com/coveo/ui-kit/blob/bff8876908fde6febaba8f9cadc0f23d199afc33/packages/headless/src/controllers/commerce/core/interactive-product/headless-core-interactive-product.test.ts#L55). That said, I understand the need to differentiate from the `purchase` method which alters state in the controller.